SCloudRandomizer/SearchParams.h

```diff
@@ -8,13 +8,15 @@
 
 #import <Foundation/Foundation.h>
 
-@interface SearchParams : NSObject
+@interface SearchParams : NSObject<NSCoding>
 
 @property BOOL hasChanged;
-@property (strong, nonatomic) NSNumber *lowBpm;
-@property (strong, nonatomic) NSNumber *highBpm;
+@property NSInteger lowBpm;
+@property NSInteger highBpm;
```

Comment:
NSNumber is sublclass of NSObject, NSInteger is primitive. According to stacoverflow its best to stick with NSInteger unless you need the object encapsulation for something.